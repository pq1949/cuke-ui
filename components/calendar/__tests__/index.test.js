import moment from "moment";
import React from "react";
import assert from "power-assert";
import { render, shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Calendar from "../index";
import Button from "../../button";

describe("<Calendar/>", () => {
  it("should render a <Calendar/> components", () => {
    const wrapper = render(<Calendar />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it("should render find .cuke-calendar classnames", () => {
    const wrapper = shallow(<Calendar />);
    assert(wrapper.find(".cuke-calendar").length === 1);
  });
  it("should render find .cuke-calendar-loading classnames", () => {
    const wrapper = shallow(<Calendar loading tip="loading" />);
    assert(wrapper.find(".cuke-calendar-loading").length === 1);
  });

  it("should window cannot trigger click event when un mount", () => {
    const onClick = jest.fn();
    const wrapper = shallow(<Calendar />);
    wrapper.unmount();
    window.onclick = () => onClick;
    expect(onClick).not.toHaveBeenCalled();
  });
  it("should trigger onChange", () => {
    const onChange = jest.fn();
    const wrapper = shallow(<Calendar onChange={onChange} />);
    wrapper
      .find(".cuke-calendar-current-month")
      .at(0)
      .simulate("click");
    expect(onChange).toHaveBeenCalled();
  });
  it("should trigger onMonthChange", () => {
    const onChange = jest.fn();
    const wrapper = shallow(<Calendar onMonthChange={onChange} />);
    wrapper
      .find(".cuke-calendar-button-icon")
      .at(0)
      .simulate("click");
    expect(onChange).toHaveBeenCalled();
  });
  it("should set next day when current month button clicked", () => {
    const wrapper = shallow(<Calendar value={moment("1996-08-25")} />);
    wrapper
      .find(".cuke-calendar-current-month")
      .at(1)
      .simulate("click");
    setTimeout(() => {
      expect(wrapper.state().momentSelected.valueOf()).toEqual(
        moment("1996-08-24").valueOf()
      );
    });
  });
  it("should set next month when current month button clicked", () => {
    const wrapper = shallow(<Calendar value={moment("1996-08-25")} />);
    wrapper
      .find(".cuke-calendar-button-icon")
      .at(1)
      .simulate("click");
    setTimeout(() => {
      expect(wrapper.state().momentSelected.valueOf()).toBe(
        moment("1996-09-25").valueOf()
      );
    });
  });
  it("should render custom data", () => {
    const wrapper = shallow(
      <Calendar
        dateCellRender={() => {
          return <Button>购物</Button>;
        }}
      />
    );
    assert(wrapper.find(Button).length > 1);
  });
  it("should set new moment selected value when did update ", () => {
    const wrapper = shallow(<Calendar value={moment()} />);
    wrapper.setProps({
      value: moment("1996-08-25")
    });
    wrapper.update();
    expect(wrapper.state().momentSelected).toEqual(moment("1996-08-25"));
  });

  it("should set custom value", () => {
    const wrapper = shallow(<Calendar defaultValue={moment("1996-01-01")} />);
    expect(wrapper.state().momentSelected).toEqual(moment("1996-01-01"));
    wrapper.setProps({
      value: moment("1996-08-25")
    });
    wrapper.update();
    expect(wrapper.state().momentSelected).toEqual(moment("1996-08-25"));
  });

  it("should return moment when calendar change", () => {
    let _currentDate = null;
    let _date = null;
    const wrapper = shallow(
      <Calendar
        onChange={(currentDate, date) => {
          _currentDate = currentDate;
          _date = date;
        }}
      />
    );
    wrapper
      .find(".cuke-calendar-item")
      .at(10)
      .simulate("click");
    expect(typeof _currentDate).toEqual("number");
    expect(typeof _date).toEqual("object");
  });
  it("should from 11 month => 10 month when last month button clicked", () => {
    const wrapper = shallow(<Calendar value={moment("2019-2-2")} />);
    wrapper
      .find(".cuke-calendar-last-month")
      .at(0)
      .simulate("click");
    expect(wrapper.state().momentSelected.month()).toEqual(1);
  });

  it("should render mini mode of calendar", () => {
    const wrapper = shallow(<Calendar miniMode />);
    assert(wrapper.find(".cuke-calendar-mini").length === 1);
  });
});
