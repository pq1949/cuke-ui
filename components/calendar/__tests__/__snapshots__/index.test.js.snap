// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Calendar/> should render a <Calendar/> components 1`] = `
<div
  class="cuke-calendar"
>
  <div
    class="cuke-calendar-header"
  >
    <svg
      class="cuke-calendar-button-icon"
      fill="none"
      height="1em"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="1em"
    >
      <polyline
        points="15 18 9 12 15 6"
      />
    </svg>
    <span
      class="cuke-calendar-date"
    >
      2019年   1月
    </span>
    <svg
      class="cuke-calendar-button-icon"
      fill="none"
      height="1em"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="1em"
    >
      <polyline
        points="9 18 15 12 9 6"
      />
    </svg>
  </div>
  <div
    class="cuke-calendar-list"
  >
    <div
      class="cuke-calendar-day-header"
    >
      <span
        class="cuke-calendar-item cuke-calendar-day-title"
      >
        一
      </span>
      <span
        class="cuke-calendar-item cuke-calendar-day-title"
      >
        二
      </span>
      <span
        class="cuke-calendar-item cuke-calendar-day-title"
      >
        三
      </span>
      <span
        class="cuke-calendar-item cuke-calendar-day-title"
      >
        四
      </span>
      <span
        class="cuke-calendar-item cuke-calendar-day-title"
      >
        五
      </span>
      <span
        class="cuke-calendar-item cuke-calendar-day-title"
      >
        六
      </span>
      <span
        class="cuke-calendar-item cuke-calendar-day-title"
      >
        日
      </span>
    </div>
    <div
      class="cuke-calendar-content"
    >
      <span
        class="cuke-calendar-item cuke-calendar-last-month"
      >
        30
      </span>
      <span
        class="cuke-calendar-item cuke-calendar-current-month"
      >
        1
        <div
          class="cuke-calendar-item-content"
        />
      </span>
      <span
        class="cuke-calendar-item cuke-calendar-current-month cuke-calendar-selected-date"
      >
        2
        <div
          class="cuke-calendar-item-content"
        />
      </span>
      <span
        class="cuke-calendar-item cuke-calendar-current-month"
      >
        3
        <div
          class="cuke-calendar-item-content"
        />
      </span>
      <span
        class="cuke-calendar-item cuke-calendar-current-month"
      >
        4
        <div
          class="cuke-calendar-item-content"
        />
      </span>
      <span
        class="cuke-calendar-item cuke-calendar-current-month"
      >
        5
        <div
          class="cuke-calendar-item-content"
        />
      </span>
      <span
        class="cuke-calendar-item cuke-calendar-current-month"
      >
        6
        <div
          class="cuke-calendar-item-content"
        />
      </span>
      <span
        class="cuke-calendar-item cuke-calendar-current-month"
      >
        7
        <div
          class="cuke-calendar-item-content"
        />
      </span>
      <span
        class="cuke-calendar-item cuke-calendar-current-month"
      >
        8
        <div
          class="cuke-calendar-item-content"
        />
      </span>
      <span
        class="cuke-calendar-item cuke-calendar-current-month"
      >
        9
        <div
          class="cuke-calendar-item-content"
        />
      </span>
      <span
        class="cuke-calendar-item cuke-calendar-current-month"
      >
        10
        <div
          class="cuke-calendar-item-content"
        />
      </span>
      <span
        class="cuke-calendar-item cuke-calendar-current-month"
      >
        11
        <div
          class="cuke-calendar-item-content"
        />
      </span>
      <span
        class="cuke-calendar-item cuke-calendar-current-month"
      >
        12
        <div
          class="cuke-calendar-item-content"
        />
      </span>
      <span
        class="cuke-calendar-item cuke-calendar-current-month"
      >
        13
        <div
          class="cuke-calendar-item-content"
        />
      </span>
      <span
        class="cuke-calendar-item cuke-calendar-current-month"
      >
        14
        <div
          class="cuke-calendar-item-content"
        />
      </span>
      <span
        class="cuke-calendar-item cuke-calendar-current-month"
      >
        15
        <div
          class="cuke-calendar-item-content"
        />
      </span>
      <span
        class="cuke-calendar-item cuke-calendar-current-month"
      >
        16
        <div
          class="cuke-calendar-item-content"
        />
      </span>
      <span
        class="cuke-calendar-item cuke-calendar-current-month"
      >
        17
        <div
          class="cuke-calendar-item-content"
        />
      </span>
      <span
        class="cuke-calendar-item cuke-calendar-current-month"
      >
        18
        <div
          class="cuke-calendar-item-content"
        />
      </span>
      <span
        class="cuke-calendar-item cuke-calendar-current-month"
      >
        19
        <div
          class="cuke-calendar-item-content"
        />
      </span>
      <span
        class="cuke-calendar-item cuke-calendar-current-month"
      >
        20
        <div
          class="cuke-calendar-item-content"
        />
      </span>
      <span
        class="cuke-calendar-item cuke-calendar-current-month"
      >
        21
        <div
          class="cuke-calendar-item-content"
        />
      </span>
      <span
        class="cuke-calendar-item cuke-calendar-current-month"
      >
        22
        <div
          class="cuke-calendar-item-content"
        />
      </span>
      <span
        class="cuke-calendar-item cuke-calendar-current-month"
      >
        23
        <div
          class="cuke-calendar-item-content"
        />
      </span>
      <span
        class="cuke-calendar-item cuke-calendar-current-month"
      >
        24
        <div
          class="cuke-calendar-item-content"
        />
      </span>
      <span
        class="cuke-calendar-item cuke-calendar-current-month"
      >
        25
        <div
          class="cuke-calendar-item-content"
        />
      </span>
      <span
        class="cuke-calendar-item cuke-calendar-current-month"
      >
        26
        <div
          class="cuke-calendar-item-content"
        />
      </span>
      <span
        class="cuke-calendar-item cuke-calendar-current-month"
      >
        27
        <div
          class="cuke-calendar-item-content"
        />
      </span>
      <span
        class="cuke-calendar-item cuke-calendar-current-month"
      >
        28
        <div
          class="cuke-calendar-item-content"
        />
      </span>
      <span
        class="cuke-calendar-item cuke-calendar-current-month"
      >
        29
        <div
          class="cuke-calendar-item-content"
        />
      </span>
      <span
        class="cuke-calendar-item cuke-calendar-current-month"
      >
        30
        <div
          class="cuke-calendar-item-content"
        />
      </span>
      <span
        class="cuke-calendar-item cuke-calendar-current-month"
      >
        31
        <div
          class="cuke-calendar-item-content"
        />
      </span>
      <span
        class="cuke-calendar-item cuke-calendar-next-month"
      >
        1
      </span>
      <span
        class="cuke-calendar-item cuke-calendar-next-month"
      >
        2
      </span>
      <span
        class="cuke-calendar-item cuke-calendar-next-month"
      >
        3
      </span>
    </div>
  </div>
</div>
`;
